src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -492,196 +543,248 @@ protected void Accepting(object message)
+                          }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default)).Unwrap().PipeTo(sender);
 
 
-                    foreach (var handoff in pendingReadHandoffs.Values)
-                    {
-                       handoff.Disassociate(DisassociateInfo.Shutdown);
-                    }
-                    
+                foreach (var handoff in _pendingReadHandoffs.Values)
+                {
+                    handoff.Disassociate(DisassociateInfo.Shutdown);
+                }
+
                     //Ignore all other writes
                     _normalShutdown = true;
-                    Context.Become(Flushing);
-                });
+                Become(Flushing);
+            });
         }
 
-        protected void Flushing(object message)
+        protected void Flushing()
         {
-            message.Match()
-                .With<Send>(send => Context.System.DeadLetters.Tell(send))
-                .With<InboundAssociation>(
-                    ia => ia.Association.AsInstanceOf<AkkaProtocolHandle>().Disassociate(DisassociateInfo.Shutdown))
-                .With<Terminated>(terminated => { });
+            Receive<Send>(send => Context.System.DeadLetters.Tell(send));
+            Receive<InboundAssociation>(
+                     ia => ia.Association.AsInstanceOf<AkkaProtocolHandle>().Disassociate(DisassociateInfo.Shutdown));
+            Receive<Terminated>(terminated => { }); // why should we care now?
         }
 
         #endregion
 
         #region Internal methods
 
-        private void HandleInboundAssociation(InboundAssociation ia)
+        private void HandleInboundAssociation(InboundAssociation ia, bool writerIsIdle)
         {
-            var readonlyEndpoint = endpoints.ReadOnlyEndpointFor(ia.Association.RemoteAddress);
-            var handle = ((AkkaProtocolHandle) ia.Association);
+            var readonlyEndpoint = _endpoints.ReadOnlyEndpointFor(ia.Association.RemoteAddress);
+            var handle = ((AkkaProtocolHandle)ia.Association);
             if (readonlyEndpoint != null)
             {
-                if (pendingReadHandoffs.ContainsKey(readonlyEndpoint)) pendingReadHandoffs[readonlyEndpoint].Disassociate();
-                pendingReadHandoffs.AddOrSet(readonlyEndpoint, handle);
-                readonlyEndpoint.Tell(new EndpointWriter.TakeOver(handle, Self));
+                var endpoint = readonlyEndpoint.Item1;
+                if (_pendingReadHandoffs.ContainsKey(endpoint)) _pendingReadHandoffs[endpoint].Disassociate();
+                _pendingReadHandoffs.AddOrSet(endpoint, handle);
+                endpoint.Tell(new EndpointWriter.TakeOver(handle, Self));
+                _endpoints.WritableEndpointWithPolicyFor(handle.RemoteAddress).Match()
+                    .With<Pass>(pass =>
+                    {
+                        pass.Endpoint.Tell(new ReliableDeliverySupervisor.Ungate());
```

Comment:
`Ungate` the `ReliableDeliverySupervisor` if we already have a `Pass` policy in place for this address.